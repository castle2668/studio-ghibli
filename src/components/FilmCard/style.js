import styled from "styled-components";

export const Card = styled.div`
  box-shadow: 0 0.4rem 0.8rem 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  &:hover {
    box-shadow: 0 0.8rem 1.6rem 0 rgba(0, 0, 0, 0.2);
  }
  cursor: pointer;
`;

export const Image = styled.img`
  display: block;
  width: 100%;
`;

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 0.8rem;
  width: 100%;
  padding: 0px 1.6rem;
`;

export const Title = styled.h4`
  font-weight: bold;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;
