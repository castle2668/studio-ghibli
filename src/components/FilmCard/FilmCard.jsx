import PropTypes from "prop-types";
import React from "react";
import { useNavigate } from "react-router-dom";

import * as Style from "./style";

const FilmCard = ({ film }) => {
  const navigate = useNavigate();

  const handleClick = () => navigate(`${film.id}`);

  return (
    <Style.Card onClick={handleClick}>
      <Style.Image src={film.image} alt="Avatar" />
      <Style.Container>
        <Style.Title>{film.title}</Style.Title>
        <p>{film.release_date}</p>
      </Style.Container>
    </Style.Card>
  );
};

FilmCard.propTypes = {
  film: PropTypes.shape({
    id: PropTypes.string,
    image: PropTypes.string,
    title: PropTypes.string,
    release_date: PropTypes.string,
  }),
};
FilmCard.defaultProps = {
  film: {},
};

export default FilmCard;
