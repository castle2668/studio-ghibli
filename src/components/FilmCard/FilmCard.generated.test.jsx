import React from "react";
import renderer from "react-test-renderer";

import FilmCard from "./FilmCard";

jest.mock("react-router-dom");
jest.mock("./style");

const renderTree = (tree) => renderer.create(tree);

describe("<FilmCard>", () => {
  it("should render component", () => {
    expect(renderTree(<FilmCard />).toJSON()).toMatchSnapshot();
  });
  it("should render component with props", () => {
    expect(
      renderTree(
        <FilmCard
          film={{
            id: "",
            image: "",
            title: "",
            release_date: "",
          }}
        />,
      ).toJSON(),
    ).toMatchSnapshot();
  });
});
