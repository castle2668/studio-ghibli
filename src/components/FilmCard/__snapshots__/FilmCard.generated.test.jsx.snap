// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmCard> should render component 1`] = `null`;

exports[`<FilmCard> should render component with props 1`] = `null`;
